sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_patch.py

```diff
@@ -6,8 +6,83 @@
 Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
 """
 
-from typing import List
+from typing import List, Union, Any, TYPE_CHECKING, Optional
+from azure.core.credentials import AzureKeyCredential, AzureSasCredential
+
+
 from .._legacy.aio import EventGridPublisherClient
+from ._client import EventGridClient as InternalEventGridClient
+from .._serialization import Deserializer, Serializer
+from .._patch import (
+    ClientLevel,
+    DEFAULT_BASIC_API_VERSION,
+    DEFAULT_STANDARD_API_VERSION,
+)
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from azure.core.credentials_async import AsyncTokenCredential
+
+
+class EventGridClient(InternalEventGridClient):
+    """Azure Messaging EventGrid Client.
+
+    :param endpoint: The endpoint to the Event Grid resource.
+    :type endpoint: str
+    :param credential: Credential needed for the client to connect to Azure.
+    :type credential: ~azure.core.credentials.AzureKeyCredential or ~azure.core.credentials.AzureSasCredential or
+     ~azure.core.credentials_async.AsyncTokenCredential
+    :keyword api_version: The API version to use for this operation. Default value for namespaces is
+     "2023-10-01-preview". Default value for basic is "2018-01-01". Note that overriding this default value may result in unsupported
+     behavior.
+    :paramtype api_version: str or None
+    :keyword level: The level of the event grid service.
```

Comment:
nit: for consistency
```suggestion
    :keyword level: The level of the Event Grid service.
```